nav2_msgs/action/ComputePathThroughPoses.action

```diff
@@ -1,5 +1,5 @@
 #goal definition
-geometry_msgs/PoseStamped[] goals
+geometry_msgs/PoseStampedArray goals
```

Comment:
what do you think of renaming `goals` to `poses`?

Reply:
I think `goals` is still a good enough name, but more I'm hesitant to change it since its used in so many places and its easy (especially w.r.t. the blackboard / BT XML / ports) to miss a few places or have users of new distributions have subtle breakages.

I'd leave it unless it was a serious bother. I still think navigating through poses' `poses` are themselves `goals`, just intermediate goals